### PR TITLE
[MRG] Expose copyfile utilities to command line interface

### LINKED ISSUE #220: CLI: Expose the "copyfile" functions as possible command line interfaces
With MNE-BIDS we offer utility to rename files from several manufacturers, e.g. BrainVision: See the [example](https://851-89170358-gh.circle-artifacts.com/0/html/auto_examples/rename_brainvision_files.html#sphx-glr-auto-examples-rename-brainvision-files-py)

These utilities are important, because the specific files cannot simply be renamed with right-click and rename: They need to be fixed internally as well.

It would be cool if we could expose these utility functions as part of our [command line interface](https://851-89170358-gh.circle-artifacts.com/0/html/index.html#command-line-interface).

one could do
1. `pip install mne_bids`
1. `mne_bids copyfile_brainvision --input myfile.vhdr --output myNEWfile.vhdr`

perhaps one could rename `copyfile_brainvision` to `rename_brainvision` ... only for the sake of more prominently advertising the purpose of the function within the CLI (under the hood, we still call `copyfile_brainvision`)

### PR DESCRIPTION
**Pull request by sappelhoff**
closes #220 

rendered docs: https://882-89170358-gh.circle-artifacts.com/0/html/generated/cli.html

# To Do:

- [x] make MNE-Python's optparse wrapper more versatile to use in mne-bids
    - [x] see: https://github.com/mne-tools/mne-python/pull/6572 ... needs to be merged
- [x] make `mne_bids cp` work for
    - BrainVision
    - CTF
    - EEGLAB format (.set / .fdt)
    - ...
-  ~~make a documentation "entry point" for mne-bids and from there, linking to:~~
    - don't do that --- cli can be linked from pagenav for now.
    - [x] examples (currently called "Gallery")
    - [x] docs page for CLI
- [x] make a documentation page for CLI
- [x] get rid of "source" and "this page" links at top of docs: They take away space

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [x] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [x] PR description includes phrase "closes <#issue-number>"
- [x] Commit history does not contain any merge commits

### CODE REVIEW

**Comment by jasmainak** [comment]
Let us know when it's ready for review

File: cli/mne_bids_cp.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                      help='Set logging level to verbose', action="store_true")
+
+    opt, args = parser.parse_args()
+    opt_dict = vars(opt)
+
+    # Check the usage and raise error if invalid
+    if len(args) > 0:
+        parser.error('Do not specify arguments without flags. Found: "{}".\n'
+                     'Did you forget to provide -i and -o?'
+                     .format(args))
```
Comment (jasmainak) [inline cli/mne_bids_cp.py]: You should not throw an error. Rather it should display the usage and help.

File: cli/mne_bids_raw_to_bids.py [hunk 2]
```diff
@@ -1,62 +1,58 @@
...
+                      help='The path to the configuration file')
     parser.add_option('--overwrite', dest='overwrite',
-                      help=("Boolean. Whether to overwrite existing data"),
-                      metavar='ow')
+                      help="Boolean. Whether to overwrite existing data")
     parser.add_option('--allow_maxshield', dest='allow_maxshield',
-                      help=("Boolean. Whether to allow non Maxfiltered data."),
-                      metavar='mxf', action='store_true')
+                      help="Boolean. Whether to allow non Maxfiltered data.",
+                      action='store_true')
```
Comment (jasmainak) [inline cli/mne_bids_raw_to_bids.py]: why did you remove the `metavar`. You shouldn't edit stuff not related to your PR.

**Comment by jasmainak** [comment]
Just one tiny nitpick, otherwise looks good!

File: README.rst [hunk 3]
```diff
@@ -99,13 +100,17 @@ iEEG is experimental at the moment.
...
+accepted commands. Then type ``mne_bids <command> --help``, where ``<command>``
+is one of the accepted commands, to get more information about that
+``<command>``.
 
 Example:
 
 .. code-block:: bash
 
-   $ mne_bids raw_to_bids --subject_id sub01 --task rest --raw data.edf --output_path new_path
+   $ mne_bids raw_to_bids --help
```
Comment (agramfort) [inline README.rst]: if you can put a few usage examples it will help a lot. It should be simpler to adapt and use than reading the --help

File: cli/mne_bids_cp.py (hunk 1 above)
Comment (sappelhoff) [reply cli/mne_bids_cp.py]: Why should it not throw an error?

This is the current output:

```Shell
(mne) stefanappelhoff@arc-lin-004309:~/Desktop/bids/mne-bids$ mne_bids cp a b
Usage: mne_bids cp -i INPUT -o OUTPUT

mne_bids cp: error: Do not specify arguments without flags. Found: "['a', 'b']".
Did you forget to provide -i and -o?

```

File: cli/mne_bids_raw_to_bids.py (hunk 2 above)
Comment (sappelhoff) [reply cli/mne_bids_raw_to_bids.py]: but what is the point of having a `metavar=r` for an argument `raw`? it is arguably more confusing to see the help showing `--raw=r` instead of `--raw=RAW`

going on along these lines, I got the idea that perhaps @teonbrooks mistook the `metavar` field for the "shortform" of an arg ... e.g., specify **RAW** either through `--raw` or `-r` ... however, to get that, you would need the following syntax (no `metavar`)

`parser.add_option('-r', '--raw', dest='raw', ...)`

I might be wrong though, and in that case - sorry for falsely suspecting Teon :-)

**Comment by agramfort** [comment]
@sappelhoff can you just see why appveyor complains?

File: cli/mne_bids_raw_to_bids.py (hunk 2 above)
Comment (jasmainak) [reply cli/mne_bids_raw_to_bids.py]: That's fine, just do it in a different PR. That's all I am saying :-) Make life of reviewer easy ...

File: cli/mne_bids_cp.py (hunk 1 above)
Comment (jasmainak) [reply cli/mne_bids_cp.py]: Because that's how most command line utils work. Try doing:

```sh
$ git
$ git mv
```

Comment (sappelhoff) [reply cli/mne_bids_cp.py]:
```
(base) stefanappelhoff@arc-lin-004309:~$ git mv
fatal: not a git repository (or any of the parent directories): .git

```

Comment (jasmainak) [reply cli/mne_bids_cp.py]: Do it in a git folder :)

Comment (jasmainak) [reply cli/mne_bids_cp.py]: But okay I can live with this. Resolving

**Comment by sappelhoff** [comment]
> @sappelhoff can you just see why appveyor complains?

I think the server was momentarily down. Let's wait for the rebuild

**Comment by agramfort** [comment]
Thx @sappelhoff

**Comment by jasmainak** [comment]
This is a great addition!
